### PR TITLE
Fix placement group creation in SkyRL-Train backend

### PR DESCRIPTION
**Pull request by tyler-griggs**
## PR Summary                                       
                                                                                                
Fix a couple issues I hit when running Tinker API server with `--backend  skyrl_train`:                                                                                 
                                                                                                
  - **Placement group fixes**: Refactor `create_model` to use `RayPPOTrainer.build_models()`, matching the logic used in model setup in `main_base.py`                                                            
  - **flashinfer version mismatch**: Pin `flashinfer-jit-cache==0.5.3` in skyrl-train's vllm    
  extra and add constraint in skyrl-tx to match vllm 0.13.0's `flashinfer-python==0.5.3`        
  requirement — mismatched versions crashed vLLM inference engine workers                       
  - **Support datasetless trainer**: Guard `train_dataset=None` in `RayPPOTrainer` so the Tinker
   backend (which provides data externally via API) can skip dataloader construction and LR     
  scheduler step calculation                                                                    
                                                                                                
  Verified working end-to-end: `sl_loop.py` from tinker-cookbook runs successfully with         
  SkyRL-Train backend.

### CODE REVIEW

File: skyrl-tx/tx/tinker/backends/skyrl_train.py [hunk 1]
```diff
@@ -212,23 +272,23 @@ def optim_step(self, model_id: str, request_data: types.OptimStepInput) -> types
...
+        grad_norm = self._trainer.dispatch.optim_step("policy")
         logger.info(f"optim_step: lr={adam_params.learning_rate}, grad_norm={grad_norm}")
         return types.OptimStepOutput()
 
     def sample(
         self,
         prepared_batch: types.PreparedSampleBatch,
     ) -> dict[str, types.SampleOutput | types.ErrorResponse]:
-        raise NotImplementedError("Sampling not supported")
+        raise NotImplementedError("Sampling not yet implemented - will be merged from tyler/tinker-sampling-main")
```
Comment (SumanthRH) [inline skyrl-tx/tx/tinker/backends/skyrl_train.py]: remove?

File: skyrl-tx/pyproject.toml [hunk 2]
```diff
@@ -101,6 +101,10 @@ required-environments = [
     "sys_platform == 'darwin' and platform_machine == 'arm64'",
 ]
 
+constraint-dependencies = [
+    "flashinfer-jit-cache==0.5.3",
```
Comment (pcmoritz) [inline skyrl-tx/pyproject.toml:105]: Do you know why this is needed?

Comment (tyler-griggs) [reply skyrl-tx/pyproject.toml:105]: It was picking up flashinfer-jit-cache==0.6.2, which would conflict with skyrl-train
